src/assignments/yunfeng/lesson_2_answers.clj

```diff
@@ -1,3 +1,23 @@
 (ns lesson-2-answers)
 
 ;;; Enter your answers in here.
+
+(require '[clojure.string :as string])
+
+(defn find-anagrams
+  [vec]
+  (->> (map (fn [str] [str str]) vec)
+       (map (fn [[key value]] {(#(clojure.string/join "" %) (sort (#(string/split % #"") key))) value}))
```

Comment:
This is hard to read and reason about what it does.  Can you break this up into more meaningful components?

Reply:
Sure. I will update the code.